plugins/guests/windows/cap/change_host_name.rb

```diff
@@ -28,7 +28,9 @@ def self.change_host_name_and_wait(machine, name, sleep_timeout)
             error_key: :rename_computer_failed)
 
           # Don't continue until the machine has shutdown and rebooted
-          sleep(sleep_timeout)
+	  if machine.guest.capability?(:wait_for_reboot)
+            machine.guest.capability(:wait_for_reboot)
```

Comment:
These two lines are the reasons for your rspec failures. You need to `allow` the machine double to be able to receive both `guest`, and the two functions `capability?` and `capability`.